core/src/sampling/RandomSampler.cpp

```diff
@@ -139,69 +139,27 @@ void RandomSampler::draw(std::vector<size_t>& result,
                          size_t max,
                          const std::set<size_t>& skip,
                          size_t num_samples) {
-  if (num_samples < max / 2) {
-    draw_simple(result, max, skip, num_samples);
-  } else {
-    draw_knuth(result, max, skip, num_samples);
-  }
-}
 
-void RandomSampler::draw_simple(std::vector<size_t>& result,
-                                size_t max,
-                                const std::set<size_t>& skip,
-                                size_t num_samples) {
-  result.reserve(num_samples);
+  // Create indices
+  result.resize(max);
+  std::iota(result.begin(), result.end(), 0);
 
-  // Set all to not selected
-  std::vector<bool> temp;
-  temp.resize(max, false);
 
-  std::uniform_int_distribution<size_t> unif_dist(0, max - 1 - skip.size());
-  for (size_t i = 0; i < num_samples; ++i) {
-    size_t draw;
-    do {
-      draw = unif_dist(random_number_generator);
-      for (auto& skip_value : skip) {
-        if (draw >= skip_value) {
-          ++draw;
-        }
-      }
-    } while (temp[draw]);
-    temp[draw] = true;
-    result.push_back(draw);
+  // Remove values that are to be skipped
+  result.erase(std::remove_if(result.begin(), result.end(),
```

Comment:
For example, I think that compared to the ranger approach, this will incur an extra linear lookup per element (through the check involving `skip.find(i)`).